app.go

```diff
@@ -220,8 +258,14 @@ func (app *App) provide(constructor interface{}) {
 		return
 	}
 	app.logger.PrintProvide(constructor)
+
+	if _, ok := constructor.(Option); ok {
+		app.optionErr = fmt.Errorf("cannot Provide fx.Option: %v", constructor)
```

Comment:
Meaning the end-user should understand what they did wrong and how to correct it just by reading the error message.